### PR TITLE
Disable logger in the child after fork (bsc#1192436)

### PR DESCRIPTION
**Pull request by bzeller**

### CODE REVIEW

**Comment by Vogtinator** [comment]
Why not use `pthread_atfork`?

File: zypp-core/zyppng/io/forkspawnengine.cc [hunk 1]
```diff
@@ -103,7 +104,9 @@ void zyppng::AbstractDirectSpawnEngine::mapExtraFds ( int controlFd )
   const auto maxFds = ( ::getdtablesize() - 1 );
   //If the rlimits are too high we need to use a different approach
   // in detecting how many fds we need to close, or otherwise we are too slow (bsc#1191324)
-  if ( maxFds >= 1024 && zypp::PathInfo( "/proc/self/fd" ).isExist() ) {
+  std::vector<int> fdsToClose;
```
Comment (Vogtinator) [inline zypp-core/zyppng/io/forkspawnengine.cc:107]: Only used within the `if`, scope can be reduced

**Comment by bzeller** [comment]
> `pthread_atfork`

Good point, will change it

File: zypp-core/zyppng/io/forkspawnengine.cc (hunk 1 above)
Comment (bzeller) [reply zypp-core/zyppng/io/forkspawnengine.cc:107]: Good point

**Pull request merged**